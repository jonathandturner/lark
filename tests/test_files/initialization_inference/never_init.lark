//~ execute:no

// FIXME: this should not be considered an unresolved variable,
// rather we should get an error about use of an initialize variable

struct Bar { x: uint }

def main() {
  let x
  //~ ERROR: Unresolved variable
  //~ ERROR: Unresolved variable
  //~ ERROR: Unresolved variable
  take(x)
  //~ ERROR: access to uninitialized path
}

def take(v: Bar) { }
