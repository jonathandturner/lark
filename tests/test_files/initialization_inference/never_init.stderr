error: access to uninitialized path
- initialization_inference/never_init:13:7
13 |   take(x)
   |        ^
error: Unresolved variable
- initialization_inference/never_init:9:2
9 |   let x

  |   ^^^^^^
error: Unresolved variable
- initialization_inference/never_init:9:2
9 |   let x

  |   ^^^^^^
error: Unresolved variable
- initialization_inference/never_init:9:2
9 |   let x

  |   ^^^^^^
